### PR TITLE
freepascal: Add version 3.2.2 #845

### LINKED ISSUE #844: [Request] Add FreePascal
`extras` bucket has `Lazarus` which is graphical IDE tool based on free pascal programming language. But often required to build a library without GUI interaction and this is impossible, since bare pascal is not available on scoop.

I believe [FreePascal](https://www.freepascal.org/) could make very nice addition to scoop's `main` bucket.

### PR DESCRIPTION
**Pull request by Valinor**
- Closes #844 

Providing definition file that will allow scoop install `freepascal` compiler (FPC) 

Thanks to https://github.com/shytikov for contribution

### CODE REVIEW

**Comment by rashil2000** [comment]
They have a 64bit version too?

image

**Comment by Valinor** [comment]
> They have a 64bit version too?
> 
> image

Ok, I will check this today.

**Comment by issaclin32** [comment]
`PullRequestHandler` fails because *SourceForge* redirects the URL [base on the user's location](https://sourceforge.net/p/forge/documentation/Mirrors/).

I think we can ignore the message for now, and fix `PullRequestHandler` later.

```
> wget.exe https://downloads.sourceforge.net/project/freepascal/Win32/3.2.2/fpc-3.2.2.i386-win32.exe
--2021-11-02 21:56:24--  https://downloads.sourceforge.net/project/freepascal/Win32/3.2.2/fpc-3.2.2.i386-win32.exe
Resolving downloads.sourceforge.net (downloads.sourceforge.net)... 204.68.111.105
Connecting to downloads.sourceforge.net (downloads.sourceforge.net)|204.68.111.105|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://nchc.dl.sourceforge.net/project/freepascal/Win32/3.2.2/fpc-3.2.2.i386-win32.exe [following]
--2021-11-02 21:56:24--  https://nchc.dl.sourceforge.net/project/freepascal/Win32/3.2.2/fpc-3.2.2.i386-win32.exe
Resolving nchc.dl.sourceforge.net (nchc.dl.sourceforge.net)... 140.110.96.69
Connecting to nchc.dl.sourceforge.net (nchc.dl.sourceforge.net)|140.110.96.69|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 53470080 (51M) [application/octet-stream]
Saving to: 'fpc-3.2.2.i386-win32.exe'

fpc-3.2.2.i386-win32.exe      100%[=================================================>]  50.99M  11.3MB/s    in 4.5s

2021-11-02 21:56:29 (11.3 MB/s) - 'fpc-3.2.2.i386-win32.exe' saved [53470080/53470080]
```

File: bucket/freepascal.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "hash": "sha1:c3fc24290da7349e4fba763eaadd2a5bec07bfa3",
+    "innosetup": true,
+    "bin": [
+        "bin\\i386-win32\\fp.exe",
+        "bin\\i386-win32\\fpc.exe"
+    ],
+    "shortcuts": [
+        [
+            "bin\\i386-win32\\fp.exe",
+            "Free Pascal IDE",
```
Comment (pratikpc) [inline bucket/freepascal.json:19]: Does it belong to Extras instead if it also has an IDE?

File: bucket/freepascal.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            "",
+            "bin\\i386-win32\\fp32.ico"
+        ]
+    ],
+    "checkver": {
+        "url": "https://sourceforge.net/p/freepascal/news/?source=navbar",
```
Comment (pratikpc) [inline bucket/freepascal.json:25]: We could replace this with https://sourceforge.net/p/freepascal/news/ but the issue is that this does not mention 3.2.0 while the version is present here https://sourceforge.net/projects/freepascal/files/Win32/3.2.0/

File: bucket/freepascal.json (hunk 1 above)
Comment (issaclin32) [reply bucket/freepascal.json:19]: I think IDE is more like an "addition" to the main program.
Similar to [latex](https://github.com/ScoopInstaller/Main/blob/master/bucket/latex.json).

**Comment by rashil2000** [comment]
@Valinor any updates?

**Comment by Valinor** [comment]
> They have a 64bit version too?

> @Valinor any updates?

Hum, this is a true problem for me, they don't have a 64bits version, they only have a 32/64 bits versions, and some 32 bits version with 64bits compiler support.

The fcp compiler stay a 32bits software and add some x86_64 tools to compile for 64bits system...
image

I don't know what people want here. But it's clearly not a 64bits version. It's a 32bits version with 64bits capabilities.

Maybe freepascal32 and freepascal64 manifest for each version?

**Comment by rashil2000** [comment]
On their download page, they recommend the i386-win32 link for both 32 and 64bit versions.

Actually, not even these binaries are 64bit - they're 32bit:

image

So I think the manifest is fine as is.

**Review by rashil2000 — APPROVED** [review]

**Review by issaclin32 — APPROVED** [review]

**Pull request merged**